app/controllers/keywords_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class KeywordsController < ApplicationController
+  before_action :ensure_authentication
+  before_action :set_csv_form
+
+  def new
+    render locals: {
+      csv_form: @csv_form
+    }
+  end
+
+  def create
+    save_result = @csv_form.save(create_params)
+
+    if save_result.class == ActiveRecord::Result
+      return redirect_to new_keywords_path, notice: t('keyword.upload_csv_successfully')
```

Comment:
as I remember the `return` doesn't halt the flow so it might call the `redirect_to new_keywords_path, alert: save_result.errors.messages[:base][0]` at the end there

Normally we do like

```
if save
  return redirect_to new_keywords_path, notice: t('keyword.upload_csv_successfully')
else
  redirect_to new_keywords_path, alert: save_result.errors.messages[:base][0]
end
```

Reply:
Fixed in 933a94f3e3a9e9f22ba218d2fe47ade125b018c8